controllers/actions.github.com/autoscalingrunnerset_controller_test.go

```diff
@@ -1465,3 +1551,76 @@ var _ = Describe("Test external permissions cleanup", func() {
 		).Should(BeTrue(), "Expected role to be cleaned up")
 	})
 })
+
+var _ = Describe("Version mismatch cleanup", func() {
+	It("Should delete autoscaling runner set on a version mismatch", func() {
+		ctx := context.Background()
+		autoscalingNS, mgr := createNamespace(GinkgoT(), k8sClient)
+
+		configSecret := createDefaultSecret(GinkgoT(), k8sClient, autoscalingNS.Name)
+
+		controller := &AutoscalingRunnerSetReconciler{
+			Client:                             mgr.GetClient(),
+			Scheme:                             mgr.GetScheme(),
+			Log:                                logf.Log,
+			ControllerNamespace:                autoscalingNS.Name,
+			DefaultRunnerScaleSetListenerImage: "ghcr.io/actions/arc",
+			ActionsClient:                      fake.NewMultiClient(),
+		}
+		err := controller.SetupWithManager(mgr)
+		Expect(err).NotTo(HaveOccurred(), "failed to setup controller")
+
+		originalVersion := build.Version
+		defer func() {
+			build.Version = originalVersion
+		}()
+		build.Version = "0.2.0"
```

Comment:
Why are we introducing a new pattern here while everywhere else we're using beforeAll / afterAll?

Reply:
Because I wanted to scope the mismatch to this exact test.
I refactored it to use BeforeAll/AfterAll, but then it looked strange in a sense that this test sets the version `"0.1.0"`, which should be deleted because build.Version is "0.2.0".

When you read the test, you can see two constants and it draws attention to it.

On the rest of them, BeforeAll/AfterAll was appropriate (in my mind) because I am using the same version both for the label, and for the `build.Version`.

The intention is to draw attention here that we are actually changing the version and that is the most important part of the test

Reply:
Maybe I shouldn't have used the separate describe for it... That would be an improvement, because this `Describe` block is only used to hold one `It` test, which is self-contained

Reply:
I'll leave this to you to make the call on, the rest if fine with me